apps/teca_cartesian_mesh_diff.cpp

```diff
@@ -0,0 +1,95 @@
+#include "teca_config.h"
+#include "teca_cf_reader.h"
+#include "teca_cartesian_mesh_reader.h"
+#include "teca_cartesian_mesh_writer.h"
+#include "teca_table_sort.h"
+#include "teca_dataset_diff.h"
+#include "teca_file_util.h"
+#include "teca_mpi_manager.h"
+#include "teca_system_interface.h"
+#include "teca_mpi.h"
+
+#include <vector>
+#include <string>
+#include <iostream>
+using namespace std;
+
+
+int main(int argc, char **argv)
+{
+    teca_mpi_manager mpi_man(argc, argv);
+    int rank = mpi_man.get_comm_rank();
+
+    teca_system_interface::set_stack_trace_on_error();
+    teca_system_interface::set_stack_trace_on_mpi_error();
+
+    if (argc < 4)
+    {
+        if (rank == 0)
+        {
+            cerr << endl
+                << "Usage:" << endl
+                << "teca_cartesian_mesh_diff [test file type] [test file] "
+                   "[reference file]" << endl
+                 << endl;
+        }
+        return -1;
+    }
+
+    // parse command line
+    unsigned int t_file_type = atoi(argv[1]);
```

Comment:
rather than having a code to name file type use `teca_file_util::extension` to get the file extension directly from the test and reference files.